node-sass build error on macOS

<!-- Please do not use this issue template to report security vulnerabilities but refer to our [security policy](https://github.com/seata/seata/security/policy). -->

- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

node-sass build error on macOS

npm ERR! Binary has a problem: Error: dlopen(/Users/runner/work/incubator-seata/incubator-seata/console/src/main/resources/static/console-fe/node_modules/node-sass/vendor/darwin-arm64-111/binding.node, 0x0001): tried: '/Users/runner/work/incubator-seata/incubator-seata/console/src/main/resources/static/console-fe/node_modules/node-sass/vendor/darwin-arm64-111/binding.node' (not a mach-o file), '/System/Volumes/Preboot/Cryptexes/OS/Users/runner/work/incubator-seata/incubator-seata/console/src/main/resources/static/console-fe/node_modules/node-sass/vendor/darwin-arm64-111/binding.node' (no such file), '/Users/runner/work/incubator-seata/incubator-seata/console/src/main/resources/static/console-fe/node_modules/node-sass/vendor/darwin-arm64-111/binding.node' (not a mach-o file)

https://github.com/apache/incubator-seata/actions/runs/10021674473/job/27700609487


### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```


### Ⅲ. Describe what you expected to happen


### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):



### Ⅴ. Anything else we need to know?


### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -149,6 +149,8 @@ jobs:
             arm64v8/ubuntu:20.04 \
             bash -exc 'apt-get update -y && \
                        apt-get install maven -y && \
+                       apt-get install -y python3 python3-pip python3-distutils && \
+                       apt-get install -y build-essential && \
                        mvn -version && \
                        mvn -T 4C clean install \
                            -Dspring-boot.version=${{ matrix.springboot }} \

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -33,15 +33,26 @@ jobs:
       - name: "Use Python 3.x"
         uses: actions/setup-python@v2
         with:
-          python-version: '3.x'
-      # step 3
+          python-version: '3.12'
+      # step 3 setuptools
+      - name: Ensure pip and setuptools
+        run: |
+          python3 -m ensurepip --upgrade
+          python3 -m pip install --upgrade pip setuptools
+      # step 4 install distutils
+      - name: Install distutils
+        run: |
+          if ! python3 -c "import distutils" &>/dev/null; then
+          python3 -m pip install --upgrade setuptools
+          fi
+      # step 6
       - name: "Set up Java JDK"
         uses: actions/setup-java@v3.12.0
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-      # step 4
-      ## step 4.1: for Ubuntu and MacOS
+      # step 7
+      ## step 7.1: for Ubuntu and MacOS
       - name: "Test with Maven on '${{ matrix.os }}' OS"
         if: matrix.os != 'windows'
         run: |
@@ -50,7 +61,7 @@ jobs:
           else
             ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           fi
-      ## step 4.2: for Windows
+      ## step 7.2: for Windows
       - name: "Build with Maven on 'windows' OS (Skip tests)"
         if: matrix.os == 'windows'
         run: | # Skip tests, because too many errors in unit-test.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -34,29 +34,23 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: '3.12'
-      # step 3 setuptools
-      - name: Ensure pip and setuptools
-        run: |
-          python3 -m ensurepip --upgrade
-          python3 -m pip install --upgrade pip setuptools
-      # step 4 install distutils
-      - name: Install distutils
-        run: |
-          if ! python3 -c "import distutils" &>/dev/null; then
-          python3 -m pip install --upgrade setuptools
-          fi
-      # step 6
+      # step 3
       - name: "Set up Java JDK"
         uses: actions/setup-java@v3.12.0
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-      # step 7
-      ## step 7.1: for Ubuntu and MacOS
+      # step 4
+      ## step 4.1: for Ubuntu and MacOS
       - name: "Test with Maven on '${{ matrix.os }}' OS"
         if: matrix.os != 'windows'
         run: |
           if [ "${{ matrix.os }}" == "macos" ]; then
+             python3 -m ensurepip --upgrade;
+             python3 -m pip install --upgrade pip setuptools;
+             if ! python3 -c "import distutils" &>/dev/null; then
+                python3 -m pip install --upgrade setuptools
+             fi
             ./mvnw -T 4C clean test -P args-for-client-test  -P arrch64 -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           else
             ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
@@ -91,22 +85,32 @@ jobs:
       - name: "Checkout"
         uses: actions/checkout@v3
       # step 2
+      - name: "Use Python 3.x"
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.12'
+      # step 3
       - name: "Set up Java JDK"
         uses: actions/setup-java@v3.12.0
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-      # step 3
-      ## step 3.1: for Ubuntu and MacOS
+      # step 4
+      ## step 4.1: for Ubuntu and MacOS
       - name: "Test with Maven on '${{ matrix.os }}' OS"
         if: matrix.os != 'windows'
         run: |
           if [ "${{ matrix.os }}" == "macos" ]; then
+             python3 -m ensurepip --upgrade;
+             python3 -m pip install --upgrade pip setuptools;
+             if ! python3 -c "import distutils" &>/dev/null; then
+                python3 -m pip install --upgrade setuptools
+             fi
             ./mvnw -T 4C clean test -P args-for-client-test  -P arrch64 -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           else
             ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           fi
-      ## step 3.2: for Windows
+      ## step 4.2: for Windows
       - name: "Build with Maven on 'windows' OS (Skip tests)"
         if: matrix.os == 'windows'
         run: | # Skip tests, because too many errors in unit-test.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -55,7 +55,7 @@ jobs:
           else
             ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           fi
-      ## step 7.2: for Windows
+      ## step 4.2: for Windows
       - name: "Build with Maven on 'windows' OS (Skip tests)"
         if: matrix.os == 'windows'
         run: | # Skip tests, because too many errors in unit-test.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -1,7 +1,7 @@
 name: "test"
 
 on:
-  push:
+  pull_request:
     branches: [ test*, "*.*.*" ]
 jobs:
   # job 1
@@ -13,9 +13,7 @@ jobs:
       matrix:
         java: [ 8, 11, 17, 21 ]
         os: [
-          ubuntu,
           macos,
-          windows, # Skip tests, because too many errors in unit-test.
         ]
         springboot: [
           2.7.18         -D spring-framework.version=5.3.31,
@@ -32,13 +30,18 @@ jobs:
       - name: "Checkout"
         uses: actions/checkout@v3
       # step 2
+      - name: "Use Python 3.x"
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      # step 3
       - name: "Set up Java JDK"
         uses: actions/setup-java@v3.12.0
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.java }}
-      # step 3
-      ## step 3.1: for Ubuntu and MacOS
+      # step 4
+      ## step 4.1: for Ubuntu and MacOS
       - name: "Test with Maven on '${{ matrix.os }}' OS"
         if: matrix.os != 'windows'
         run: |
@@ -47,7 +50,7 @@ jobs:
           else
             ./mvnw -T 4C clean test -P args-for-client-test -Dspring-boot.version=${{ matrix.springboot }} -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           fi
-      ## step 3.2: for Windows
+      ## step 4.2: for Windows
       - name: "Build with Maven on 'windows' OS (Skip tests)"
         if: matrix.os == 'windows'
         run: | # Skip tests, because too many errors in unit-test.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -1,7 +1,7 @@
 name: "test"
 
 on:
-  pull_request:
+  push:
     branches: [ test*, "*.*.*" ]
 jobs:
   # job 1
@@ -13,7 +13,9 @@ jobs:
       matrix:
         java: [ 8, 11, 17, 21 ]
         os: [
+          ubuntu,
           macos,
+          windows, # Skip tests, because too many errors in unit-test.
         ]
         springboot: [
           2.7.18         -D spring-framework.version=5.3.31,

--- a/changes/en-us/2.1.0.md
+++ b/changes/en-us/2.1.0.md
@@ -52,6 +52,8 @@ The version is updated as follows:
 - [[#6555](https://github.com/apache/incubator-seata/pull/6555)] businessActionContext is compatible with io seata
 - [[#6553](https://github.com/apache/incubator-seata/pull/6553)] fix saga "cannot matching status"
 - [[#6575](https://github.com/apache/incubator-seata/pull/6575)] fix io.seata ActionInterceptorHandler use org.apache.seata BusinessActionContextParameter
+- [[#6675](https://github.com/apache/incubator-seata/pull/6675)] fix Alibaba Dubbo convert error
+- [[#6676](https://github.com/apache/incubator-seata/pull/6676)] fix hsf ConsumerModel convert error
 
 ### optimize:
 - [[#6031](https://github.com/apache/incubator-seata/pull/6031)] add a check for the existence of the undolog table
@@ -161,6 +163,12 @@ The version is updated as follows:
 - [[#6610](https://github.com/apache/incubator-seata/pull/6610)] revised the notice file
 - [[#6649](https://github.com/apache/incubator-seata/pull/6649)] add license header
 - [[#6688](https://github.com/apache/incubator-seata/pull/6688)] delete static code built on the frontend
+- [[#6670](https://github.com/apache/incubator-seata/pull/6670)] add license header for proto file
+- [[#6671](https://github.com/apache/incubator-seata/pull/6671)] optimize multiple licenses and remove license urls
+- [[#6672](https://github.com/apache/incubator-seata/pull/6672)] update front-end license
+- [[#6673](https://github.com/apache/incubator-seata/pull/6673)] add license header for spring config file
+- [[#6674](https://github.com/apache/incubator-seata/pull/6674)] update source license
+- [[#6677](https://github.com/apache/incubator-seata/pull/6677)] add license header for SPI file
 
 
 ### security:

--- a/changes/en-us/2.1.0.md
+++ b/changes/en-us/2.1.0.md
@@ -169,6 +169,7 @@ The version is updated as follows:
 - [[#6673](https://github.com/apache/incubator-seata/pull/6673)] add license header for spring config file
 - [[#6674](https://github.com/apache/incubator-seata/pull/6674)] update source license
 - [[#6677](https://github.com/apache/incubator-seata/pull/6677)] add license header for SPI file
+- [[#6692](https://github.com/apache/incubator-seata/pull/6692)] fix node-gyp build error on arm64 and macos
 
 
 ### security:

--- a/changes/zh-cn/2.1.0.md
+++ b/changes/zh-cn/2.1.0.md
@@ -168,6 +168,7 @@ Apache Seata(incubating) 是一款开源的分布式事务解决方案，提供�
 - [[#6673](https://github.com/apache/incubator-seata/pull/6673)] 为spring 配置文件添加 license header
 - [[#6674](https://github.com/apache/incubator-seata/pull/6674)] 更新源码中的 license
 - [[#6677](https://github.com/apache/incubator-seata/pull/6677)] 为 SPI配置文件添加 license header
+- [[#6692](https://github.com/apache/incubator-seata/pull/6692)] 修复 node-gyp 在 arm64 和 macos 构建失败问题
 
 
 ### security:

--- a/changes/zh-cn/2.1.0.md
+++ b/changes/zh-cn/2.1.0.md
@@ -52,6 +52,8 @@ Apache Seata(incubating) 是一款开源的分布式事务解决方案，提供�
 - [[#6555](https://github.com/apache/incubator-seata/pull/6555)] 修复businessActionContext对io seata包的不兼容
 - [[#6553](https://github.com/apache/incubator-seata/pull/6553)] 修复执行完 'ServiceTask' 后无法应用任何评估器的问题
 - [[#6575](https://github.com/apache/incubator-seata/pull/6575)] 修复io.seata ActionInterceptorHandler误使用了org.apache.seata BusinessActionContextParameter类的问题
+- [[#6675](https://github.com/apache/incubator-seata/pull/6675)] 吸修复 Alibaba Dubbo 转换错误问题
+- [[#6676](https://github.com/apache/incubator-seata/pull/6676)] 修复 hsf ConsumerModel 转换错误问题
 
 ### optimize:
 - [[#6031](https://github.com/apache/incubator-seata/pull/6031)] 添加undo_log表的存在性校验
@@ -160,6 +162,12 @@ Apache Seata(incubating) 是一款开源的分布式事务解决方案，提供�
 - [[#6610](https://github.com/apache/incubator-seata/pull/6610)] 订正  notice 文件
 - [[#6649](https://github.com/apache/incubator-seata/pull/6649)] 增加license header
 - [[#6688](https://github.com/apache/incubator-seata/pull/6688)] 删除前端构建的静态代码
+- [[#6670](https://github.com/apache/incubator-seata/pull/6670)] 为 proto 文件添加 license header
+- [[#6671](https://github.com/apache/incubator-seata/pull/6671)] 优化多 license和移除license url
+- [[#6672](https://github.com/apache/incubator-seata/pull/6672)] 更新前端 license
+- [[#6673](https://github.com/apache/incubator-seata/pull/6673)] 为spring 配置文件添加 license header
+- [[#6674](https://github.com/apache/incubator-seata/pull/6674)] 更新源码中的 license
+- [[#6677](https://github.com/apache/incubator-seata/pull/6677)] 为 SPI配置文件添加 license header
 
 
 ### security:

